modin/pandas/indexing.py

```diff
@@ -391,12 +391,16 @@ def _broadcast_item(self, row_lookup, col_lookup, item, to_shape):
                     "an iterable"
                 )
             if hasattr(item, "columns"):
-                if not all(idx in item.columns for idx in col_lookup):
+                own_cols = self.qc.columns
+                # convert indices in column lookup to column names, as Pandas reindex expects them so
+                lookup = [own_cols[idx] for idx in col_lookup]
```

Comment:
we probably should do the same changes for indices, because reindex expects index labels rather than its numeric indices, the same as it excepts the column labels.

New added test passes because in that particular case numeric indices equal to index labels, but if we'll try something like this it will crash:
```python
import modin.pandas as pd
import numpy as np

columns = ["col1", "col2", "col3"]
index = ["a", "b", "c", "d"]
df = pd.DataFrame({col: np.arange(4) for col in columns}, index=index)

df.loc[["a", "b"], :] = df.loc[["a", "b"], :].copy() # ValueError
```

Reply:
Nice, added to the test. Now it fails 😆